File: src/main/java/com/springai/semanticbooksearchlive/model/Book.java
package com.springai.semanticbooksearchlive.model;

public record Book(String id, String title, String author, String summary, String genre, int publicationYear,
                String imageUrl, boolean hasContent, java.util.List<String> tableOfContents, String textUrl) {
        public Book(String id, String title, String author, String summary, String genre, int publicationYear,
                        String imageUrl) {
                this(id, title, author, summary, genre, publicationYear, imageUrl, false, java.util.List.of(), null);
        }
}

File: frontend/src/types/Book.ts

export interface Book {
    id: string;
    title: string;
    author: string;
    summary: string;
    genre: string;
    publicationYear: number;
    imageUrl?: string;
    hasContent: boolean;
}

export interface SearchResult {
    book: Book;
    score: number;
}

export interface HybridSearchResponse {
    vectorResults: SearchResult[];
    keywordResults: SearchResult[];
    hybridResults: SearchResult[];
}
